### PR TITLE
Release 1.25.1

### PR DESCRIPTION
**Pull request by bborman22**
### Changes in this release
#2220 Lookup carrier name for email tracking URL
#2209 Add husky lint-staged for pre commit hooks
#2219 Adjust DHL refund days copy
#2223 Disable refund button for USPS letters
#2212 Add phpcs with WooCommerce-Core rules to pre-commit hook
#2228 Use is_offline_mode instead of deprecated is_development_mode

This release branch is branched from latest `develop` 

### Test notes

- Test label purchase flow
- Test live rates
- Test Store on WP.com
- Test DHL refund days
- Test USPS letter refunds
- Test pre-commit and pre-push hooks
- Test shipping carrier name in customer email notification
- Test Jetpack offline mode

### Changelog
* Tweak - DHL refund days copy adjustment
* Tweak - Stop using deprecated Jetpack method is_development_mode().
* Fix       - Update carrier name in tracking notification email
* Add     - Add pre-commit and pre-push git hooks for linting and unit tests.
* Add     - Disable refunds for USPS letters.

## Build File
Built from this branch using `php woorelease.phar simulate --product_version=1.25.1 https://github.com/Automattic/woocommerce-services/tree/release/1.25.1`

[woocommerce-services.zip](https://github.com/Automattic/woocommerce-services/files/5430408/woocommerce-services.zip)

### CODE REVIEW

**Review by frosso — APPROVED** [review]
Tested against staging server environment

✅  USPS box label purchase/refund
✅  USPS letter label purchase/refund
✅  DHL label purchase/refund
✅  USPS live rates
✅  DHL live rates

File: readme.txt [hunk 1]
```diff
@@ -84,6 +84,13 @@ As of the WooCommerce 3.5 release, WooCommerce Shipping & Tax no longer provides
 
 == Changelog ==
 
+= 1.25.1 - 2020-XX-XX =
```
Comment (c-shultz) [inline readme.txt:87]: Totally not a problem on this release, but one thing to note here is that it's good to keep an eye on the length of the readme.txt so the length doesn't get out of control. The changelog is displayed here: https://wordpress.org/plugins/woocommerce-services/#developers

Every couple releases, I usually end up double-checking the length and trimming off some of the oldest entries.

**Review by c-shultz — APPROVED** [review]
Looks good to me!

Checks on my end:
✅ `1.25.1` version correct
✅ USPS package purchase
✅ USPS package refund
✅ USPS letter purchase
✅ USPS letter refund (not allowed, as expected)
✅ DHL purchase
✅ DHL refund
✅ Automated taxes
✅ USPS live rates

Most checks were made against the staging server, except taxes and live rates were tested with the production server.

Totally unrelated to this release, I noticed some awkward input validation behavior and opened an issue: https://github.com/Automattic/woocommerce-services/issues/2232

@bborman22, I added you as a committer of the plugin on WordPress.org so you should have the access you need when it comes time to release.

File: readme.txt (hunk 1 above)
Comment (bborman22) [reply readme.txt:87]: Great thanks for this info, I saw it got trimmed last time so I'll keep it in mind for the future releases.

**Comment by harriswong** [comment]
✅ USPS Letter purchase 
✅ USPS Letter not refundable
✅ Tested against store, purchase and refund DHL ok
✅ Tested UPS purchase/refund 

LGTM 👍

**Comment by bborman22** [comment]
Since everything looks good, I'll be merging into master, do one more simulated release and smoke test, then run the real release process. Once confirmed I'll be communicating the release on https://href.li/?87RIY-p2 and #woo-announcements!

**Pull request merged**